mediatek: add model files for uc22 dangerous and stable

talhaHavadar opened the pull request:
Adding mediatek ubuntu core 22 model files, using subfolder structure as discussed before.


cc: @alfonsosanchezbeato

talhaHavadar commented:
adding @EthanHsieh for sanity check

alfonsosanchezbeato reviewed (commented):
Thanks, I am thinking that it would be nice to standardize in the assertions we have for different systems. My suggestion is to use 2 assertions, similar to these 2:

- https://github.com/canonical/models/blob/master/ubuntu-core-24-arm64.json
- https://github.com/canonical/models/blob/master/ubuntu-core-24-arm64-dangerous.json

The first one is similar to the `-stable` that you propose, but please change the name to remove that suffix.

The second one is similar to the `-dangerous` one, but using the `edge` channels. Would it be possible to change yours to match this one? Note that when using `ubuntu-image` for dangerous models you can override the used channels so you can still use the dangerous assertion with stable channels. Note also that there is an exception, the default kernel channel is `24/beta` because the one in edge is not signed by a Canonical key. I'm not sure if this is the same for these devices.

talhaHavadar commented:
Thank you for the information @alfonsosanchezbeato I updated `-stable` one as you suggested, removed the suffix. 

For `-dangerous`, I think we need to resign it when we change channels in model assertion file. So I removed them from this PR and will create another one for `-dangerous`
